code/game/machinery/item_transporter.dm

```diff
@@ -0,0 +1,156 @@
+	var/mb_rating = 0
+	var/l_rating = 0
+	cooldown_time = 40 SECONDS
+	spawn_cooldown = 5 SECONDS
+	for(var/obj/item/weapon/stock_parts/matter_bin/spMatterBin in component_parts)
+		mb_rating += spMatterBin.rating
+	for(var/obj/item/weapon/stock_parts/micro_laser/spLaser in component_parts)
+		l_rating += spLaser.rating
+	cooldown_time -= (mb_rating * 10) SECONDS
+	spawn_cooldown -= l_rating SECONDS
+
+/obj/machinery/item_transporter/attack_ghost(mob/user)
+	if(isAdminGhost(user))
+		var/choice_list = list(
+			"Reset Cooldown",
+			"Set Cooldown"
+		)
+		var/choice = input(user, "Choose which action to do") as null|anything in choice_list
+		switch(choice)
+			if("Reset Cooldown")
+				timed_cooldown = 0
+				return
+			if("Set Cooldown")
+				var/num_choice = input(user, "How many seconds should the cooldown be?") as num
+				if(!num_choice)
+					return
+				if(num_choice <= 0)
+					num_choice = 1
+				cooldown_time = num_choice SECONDS
+				return
+	var/obj/machinery/item_transporter/tele_choice = input(user, "Which transporter would you like to go to?") as null|anything in current_transporters
+	if(!tele_choice)
+		return
+	user.forceMove(get_turf(tele_choice))
+
+/obj/machinery/item_transporter/attack_hand(mob/user, ignore_brain_damage)
+	if(isobserver(user))
+		return
+	if(user.incapacitated() || !user.Adjacent(src))
+		return
+	if(decon_stage)
+		to_chat(user, "<span class='warning'>\The [src] has its wire panel open, close it before using.</span>")
+		return
+	if(world.time < timed_cooldown)
+		var/time_remaining = round((timed_cooldown - world.time) / 10)
+		to_chat(user, "<span class='warning'>\The [src] is currently on cooldown, please wait [time_remaining] seconds.</span>")
+		return
+	if(busy)
+		return
+	busy = TRUE
+	var/obj/machinery/item_transporter/choice = input(user, "Which transporter will you select?") as null|anything in current_transporters
+	if(!choice)
+		busy = FALSE
+		return
+	if(user.incapacitated() || !user.Adjacent(src))
+		busy = FALSE
+		return
+	var/turf/src_turf = get_turf(src)
+	var/turf/choice_turf = get_turf(choice)
+	if(src_turf.z != choice_turf.z)
+		visible_message("<span class='notice'>\The [src] pings 'Target is too far to choose.'</span>")
+		busy = FALSE
+		return
+	visible_message("<span class='notice'>\The [src] begins to power up...</span>")
+	flick(icon, "transporter01")
+	spawn(spawn_cooldown)
+		icon_state = "transporter1"
+		for(var/obj/moveO in src_turf)
+			if(moveO.anchored)
+				continue
+			moveO.forceMove(choice_turf)
+		flick(icon, "transporter10")
+		icon_state = "transporter0"
+		visible_message("<span class='notice'>\The [src] begins to cool down...</span>")
+		timed_cooldown = world.time + cooldown_time
+		playsound(src, 'sound/machines/chime.ogg', 50, 1)
+		choice.timed_cooldown = world.time + choice.cooldown_time
+		playsound(choice, 'sound/machines/chime.ogg', 50, 1)
+		busy = FALSE
+
+/obj/machinery/item_transporter/attackby(obj/item/weapon/W as obj, mob/user as mob)
```

Comment:
we have machine flags for these behaviors -- you want screwtoggle and crowdestroy (probably wrenchanchor too)

Reply:
okay, so screwtoggle for sure, would crowdestroy put down the circuitboard? because I want to avoid that.

Reply:
yeah, crowdestroy drops all the stock parts; if you want to make it not do that you can actually override the `spillContents` proc (see machinery.dm L475)

Reply:
ohh, okay!

Reply:
also, definitely do not want wrenchanchor, as I want it always anchored (unless they deconstruct it off course)